plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java

```diff
@@ -71,10 +72,12 @@ private List<String> toConjuncts(List<BigQueryColumnHandle> columns)
         }
         ImmutableList.Builder<String> clauses = ImmutableList.builder();
         for (BigQueryColumnHandle column : columns) {
-            Domain domain = tupleDomain.getDomains().get().get(column);
-            if (domain != null) {
-                toPredicate(toBigQueryColumnName(column.getName()), domain, column).ifPresent(clauses::add);
-            }
+            tupleDomain.getDomains()
+                    .filter(domains -> domains.containsKey(column))
+                    .map(domains -> domains.get(column))
+                    .filter(Objects::nonNull)
+                    .flatMap(domain -> toPredicate(toBigQueryColumnName(column.getName()), domain, column))
+                    .ifPresent(clauses::add);
```

Comment:
Actually, I just realized that this is not needed even though it is more correct. Columns are derived from the `TupleDomains`. Given that we only reach this if domains are not empty (we have columns) and it this point, it's safe to call `.get`.